docs/current_docs/cookbook/snippets/execute-concurrently/go/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"dagger/my-module/internal/dagger"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type MyModule struct{}
+
+// Return the result of running unit tests
+func (m *MyModule) Test(ctx context.Context, source *dagger.Directory) (string, error) {
```

Comment:
If it's not too many concepts all at once, I'd probably use a constructor for the `source *dagger.Directory` here, then it doesn't need to be part of each and every function call which might make the example a bit easier to follow.